components/date-picker/picker.component.ts

```diff
@@ -343,13 +346,13 @@ export class NzPickerComponent implements OnInit, AfterViewInit, OnChanges, OnDe
   }
 
   showClear(): boolean {
-    return !this.disabled && !this.isEmptyValue(this.datePickerService.value) && !!this.allowClear;
+    return !this.inputReadOnly && !this.disabled && !this.isEmptyValue(this.datePickerService.value) && !!this.allowClear;
   }
 
   onClickInputBox(event: MouseEvent, partType?: RangePartType): void {
     event.stopPropagation();
 
-    if (!this.disabled && !this.isOpenHandledByUser()) {
+    if (!this.inputReadOnly && !this.disabled && !this.isOpenHandledByUser()) {
```

Comment:
inputReadOnly 只是为了手机端打开时不弹出键盘。打开和 clear 的行为不受影响。
```suggestion
    if (!this.disabled && !this.isOpenHandledByUser()) {
```